src/lib.rs

```diff
@@ -29,32 +31,36 @@ impl Drop for IRShaderReflection {
 }
 
 impl IRShaderReflection {
-    pub fn new(compiler: &IRCompiler) -> Result<IRShaderReflection, Box<dyn std::error::Error>> {
+    pub fn new(compiler: &IRCompiler) -> IRShaderReflection {
         unsafe {
             let me = compiler.funcs.IRShaderReflectionCreate();
-            Ok(Self {
+            Self {
                 me,
                 funcs: compiler.funcs.clone(),
-            })
+            }
         }
     }
 
     pub fn get_compute_info(
```

Comment:
```suggestion
    pub fn compute_info(
```
https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

Reply:
I'm assuming this goes for all the other getters as well? (get_reflection, get_type)